packages/unhead/src/utils/walkResolver.ts

```
@@ -1,21 +1,47 @@
-      if (k === '__proto__' || k === 'constructor' || k === 'prototype')
+      // never read/assign dangerous keys; the `continue` keeps us from ever doing
+      // `next['__proto__'] = …` (which would set a prototype). Object spread below is
+      // [[CreateDataProperty]], so it cannot pollute, and normalizeProps strips any
+      // dangerous key it carries through.
+      if (isUnsafeKey(k))
         continue
-      next[k] = walkResolver(v[k], resolve, k)
+      const r = walkResolver(v[k], resolve, k)
+      // first changed child: shallow-clone once, then keep overwriting in place
+      if (r !== v[k])
+        (next ??= { ...v })[k] = r
```

Biome (2.5.1)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)